packages/liveblocks-core/src/client.ts

```diff
@@ -654,3 +655,21 @@ function toURLSearchParams(
   }
   return result;
 }
+
+const CacheStoreMap = new WeakMap<
+  Client,
+  ReturnType<typeof createClientStore<any>>
+>();
+
+export function getCacheStore<TThreadMetadata extends BaseMetadata>(
+  client: Client
+) {
+  let store = CacheStoreMap.get(client);
```

Comment:
@GuillaumeSalles I started moving the cache store from `room` to `client` so we could also start working on other hooks like `useInboxNotifications` that fetch data from outside the room. I've gone with the `WeakMap` approach instead of adding a hidden field to `client`. What are your thoughts on this?